tools/devtool

```diff
@@ -475,14 +475,22 @@ cmd_build() {
         cargo_args+=(--target "$target")
     fi
 
-    [ $profile = "release" ] && cargo_args+=("--release")
+    machine=`uname -m`
+    # On aarch64 musl depends on some libgcc functions (i.e `__addtf3` and other `*tf3` functions) for logic that uses
+    # long double. Such functions are not builtin in the rust compiler, so we need to get them from libgcc.
+    # No need for the `crt_static` flag as rustc appends it by default.
+    [ $machine = "aarch64" ] && RUSTFLAGS="-C link-arg=-lgcc -C link-arg=-lfdt"
+    # Strip away debug info, since cargo adds them by default event when building artifacts for release.
+    # See: https://github.com/rust-lang/cargo/issues/4122.
+    [ $profile = "release" ] && cargo_args+=("--release") && RUSTFLAGS="$RUSTFLAGS -C link-arg=-s"
```

Comment:
I would leave it to the user to decide whether or not they want symbols. We might need the unstripped binary for debugging, and with this change the user can't force a build with symbols. How about a new devtool parameter (`devtool build --release --keepsyms` or smth) that lets us to both?

Reply:
Good idea. To not break things, we might want to leave the default behavior to be as it is and provide a flag `--strip_debug_info`, to enforce debug info stripping.

Reply:
I would not bother modifying the `devtool `for this since users always have the choice of using "cargo build --release" if they want the unaltered version of what cargo obtains.

Reply:
This removes all symbols, not only debug symbols. This means stack traces would be difficult to read. Keeping a small subset of symbols via `strip --discard-locals --strip-debug` would allow you to remove most symbols without breaking stack traces. It also means you don't have to change the linking, which doesn't work the same on all toolchains.

Reply:
> I would not bother modifying the devtool for this since users always have the choice of using "cargo build --release" if they want the unaltered version of what cargo obtains.
> I would not bother modifying the devtool for this since users always have the choice of using "cargo build --release" if they want the unaltered version of what cargo obtains.

@dianpopa , do you suggest to implement stripping support as the default way of building the release binaries? Now I would prefer adding _optional_ support for stripping, when building with `tools/devtool build --release`, rather than adding by default stripping support while being open to break customers. Using `cargo build --release` to get unstripped binaries depends on setting up the rust dev environment 

I see two ways for implementing this, but I prefer the first one, since it involves the minimum changes to devtool which will not break customers or force them to install any dependency on their machines, to be able to build Firecracker like before:
* adding a flag to devtool, which strips away debug info would not break customers and will provide optional support for stripping debug info.
* stripping default info by default, when building the release binaries with devtool, will drop the support for building release binaries the "cargo" way (with debug info). If user need to build release binaries with `cargo build --release` they would need to set up rust dev environment (rust, cargo, rustup, toolchains etc).

> This removes all symbols, not only debug symbols. This means stack traces would be difficult to read. [...] 

@petreeftime , stripping only what is necessary is a valid point. I wasn't aware of selective stripping until now. I'll try to incorporate your suggestion. 

> It also means you don't have to change the linking, which doesn't work the same on all toolchains.

This argument conflicts with the suggestion to use `strip`. `strip` is part of the `GNU binutils`, which currently provides `GNU ld` and `GNU as`, used by `tools/devtool build --release`. So even if we change the toolchain, we can not say that we move away from GNU binutils (this is a core part of Linux toolchains in general). To me it seems that `GNU ld` and `GNU as` (GNU binutils in general) are core parts of Linux toolchains and changing stripping alternative from linker flags to strip tool it shouldn't make any difference while `GNU binutils` is still used as a core part of the toolchain we build against.